lib/ansible/plugins/loader.py

```diff
@@ -51,7 +51,7 @@ def add_all_plugin_dirs(path):
                 if os.path.isdir(plugin_path):
                     obj.add_directory(to_text(plugin_path))
     else:
-        display.warning("Ignoring invalid path provided to plugin path: %s is not a directory" % to_native(path))
+        display.warning("Ignoring invalid path provided to plugin path: '%s' is not a directory" % to_native(path))
```

Comment:
All ```display.*``` calls need to be given text strings.

Reply:
A non-ascii byte string on python2 will trigger a traceback here whereas text strings will not:
``` python
>>> from ansible.utils.display import Display
>>> Display().warning(b'caf\xc3\xa9')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/srv/ansible/abadger/lib/ansible/utils/display.py", line 244, in warning
    self.display(new_msg, color=C.COLOR_WARN, stderr=True)
  File "/srv/ansible/abadger/lib/ansible/utils/display.py", line 131, in display
    msg = stringc(msg, color)
  File "/srv/ansible/abadger/lib/ansible/utils/color.py", line 93, in stringc
    return u"\n".join([u"\033[%sm%s\033[0m" % (color_code, t) for t in text.split(u'\n')])
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 15: ordinal not in range(128)
>>> Display().warning(b'caf\xc3\xa9'.decode('utf-8'))
 [WARNING]: café
```

Reply:
going to leave as is in current PR as this issue already existed, but creating new PR to fix this in general